src/common/string/filter/filter.ts

```diff
@@ -104,42 +115,22 @@ function isPatternInWord(
 }
 
 enum Arrow {
-  Top = 0b1,
-  Diag = 0b10,
-  Left = 0b100,
+  Diag = 1,
+  Left = 2,
+  LeftLeft = 3,
 }
 
 /**
- * A tuple of three values.
+ * An array representating a fuzzy match.
+ *
  * 0. the score
- * 1. the matches encoded as bitmask (2^53)
- * 2. the offset at which matching started
+ * 1. the offset at which matching started
+ * 2. `<match_pos_N>`
+ * 3. `<match_pos_1>`
+ * 4. `<match_pos_0>` etc
  */
-export type FuzzyScore = [number, number, number];
-
-interface FilterGlobals {
-  _matchesCount: number;
-  _topMatch2: number;
-  _topScore: number;
-  _wordStart: number;
-  _firstMatchCanBeWeak: boolean;
-  _table: number[][];
-  _scores: number[][];
-  _arrows: Arrow[][];
-}
-
-function initGlobals(): FilterGlobals {
-  return {
-    _matchesCount: 0,
-    _topMatch2: 0,
-    _topScore: 0,
-    _wordStart: 0,
-    _firstMatchCanBeWeak: false,
-    _table: initTable(),
-    _scores: initTable(),
-    _arrows: <Arrow[][]>initTable(),
-  };
-}
+// export type FuzzyScore = [score: number, wordStart: number, ...matches: number[]];// [number, number, number];
+export type FuzzyScore = Array<number>;
```

Comment:
I would prefer to use the commented out line:

```
// export type FuzzyScore = [score: number, wordStart: number, ...matches: number[]];// [number, number, number];
```
But that line doesn't pass the commit pre-check:

![image](https://user-images.githubusercontent.com/1480827/113357052-211c9c80-92f8-11eb-8129-2395630cd9b0.png)

I tried changing to:

```
export type FuzzyScore = [number, number, number[]];// [number, number, number];
```

But then typescript complains when the code doesn't include a 3rd number:

![image](https://user-images.githubusercontent.com/1480827/113357121-49a49680-92f8-11eb-93be-128c9156e8be.png)